pennylane/ops/qubit/state_preparation.py

```diff
@@ -176,6 +172,43 @@ def compute_decomposition(state, wires):
         """
         return [MottonenStatePreparation(state, wires)]
 
+    def state_vector(self, wire_order=None):
```

Comment:
General question about the device rework: what do we want to do about batching? Because this function should be compatible with it at the end of the rework. In my opinion it is easier to make sure to tackle this issue in each separate PR instead of coming back to it later. @albi3ro

Reply:
Exactly. Batching is a "nice to have".  If it would be straightforward to implement right now, go ahead and add it.  But I'd like to try and get as much basic behaviour in as possible right now and then come back to add more feature coverage.